Assets/_Laz/Scripts/Managers/CheckPointManager.cs

```diff
@@ -1,28 +1,70 @@
-using System.Collections.Generic;
+using PerigonGames;
 using UnityEngine;
 
 namespace Laz
 {
     public class CheckPointManager : MonoBehaviour
     {
-        [SerializeField] private List<Transform> _checkPoints = null;
-        public Transform LastCheckPoint { get; private set; } = null;
+        [SerializeField] private Checkpoint[] _checkpoints;
+        private Checkpoint _activeCheckpoint = null;
+        private Vector3 _initialCheckPointPosition = Vector3.zero;
+        private LazPlayer _laz = null;
 
-        public void Initialize()
+        public void Initialize(LazPlayer laz)
         {
-            UpdateCheckPoint();
+            if (_checkpoints.IsNullOrEmpty())
+            {
+                Debug.LogError("No Checkpoints Found!");
+                return;
+            }
+
+            _laz = laz;
+            
+            for (int i = 0; i < _checkpoints.Length; i++)
+            {
+                Checkpoint checkpoint = _checkpoints[i];
+
+                if (checkpoint == null)
+                {
+                    Debug.LogError("Null Checkpoint");
+                    continue;
+                }
+                
+                checkpoint.OnCheckpointActivation += SetNewCheckpoint;
+
+                if (i == 0)
+                {
+                    _activeCheckpoint = checkpoint;
+                    _activeCheckpoint.IsActiveCheckpoint = true;
+                    _initialCheckPointPosition = _activeCheckpoint.transform.position;
+                    _laz.SetSpawn(_initialCheckPointPosition);
+                }
+            }
+        }
+        
+        public Vector3 GetActiveCheckpointPosition()
+        {
+            return _activeCheckpoint != null ? _activeCheckpoint.transform.position : _initialCheckPointPosition;
+        }
+        
+        public void Reset()
+        {
+            _laz.SetSpawn(_activeCheckpoint != null ? _activeCheckpoint.transform.position : _initialCheckPointPosition);
```

Comment:
I don't think you need to reset the Laz Spawn Point here. 

because now, if we were to Reset it here, Then in WorldManager you must 
1. Reset CheckpointManager First.
2. Then Reset Laz - (since this makes Laz dependent on this line of code)